runner_manager/backend/aws.py

```diff
@@ -96,52 +92,56 @@ def list(self) -> List[Runner]:
                         "Values": ["running"],
                     },
                 ]
-            ).get("Reservations", [])
+            ).get("Reservations")
         except Exception as e:
+            log.error(e)
             raise e
         for reservation in reservations:
             for instance in reservation.get("Instances", []):
+                instance_id = instance.get("InstanceId", "")
                 try:
                     runners.append(
                         Runner.find(
-                            Runner.instance_id == instance.get("InstanceId", "")
+                            Runner.instance_id == instance_id,
                         ).first()
                     )
                 except NotFoundError:
                     runner = Runner(
-                        name=instance.get("InstanceId", ""),
-                        instance_id=instance.get("InstanceId", ""),
+                        name=instance_id,
+                        instance_id=instance_id,
                         busy=False,
                     )
                     runners.append(runner)
         return runners or []
 
     def update(self, runner: Runner) -> Runner:
         """Update a runner."""
-        try:
-            if runner.instance_id:
+        if runner.instance_id:
+            try:
                 self.client.describe_instances(
                     InstanceIds=[runner.instance_id],
                     Filters=[
                         {"Name": "instance-state-name", "Values": ["running"]},
                         {"Name": "tag-key", "Values": ["runner-manager"]},
                     ],
                 )
-        except ClientError as e:
-            error = e.response.get("Error", {})
-            if error.get("Code") == "InvalidInstanceID.NotFound":
-                log.error(f"Instance {runner.instance_id} not found.")
-            else:
-                raise e
-        try:
-            if runner.labels and runner.instance_id:
+            except ClientError as e:
+                error = e.response.get("Error", {})
+                if error.get("Code") == "InvalidInstanceID.NotFound":
+                    log.error(f"Instance {runner.instance_id} not found.")
+                    return super().update(runner)
+                else:
+                    raise e
+        if runner.labels and runner.instance_id:
```

Comment:
This could be nested in the already existing `if runner.instance_id` statement from above:

```suggestion
            if runner.labels:
```